PeerioTechnologies/peerio-desktop
# style: new designs for creating room and DM

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### PR author

#### Relevant issue tracker links  
https://peerio.tpondemand.com/entity/7469
https://peerio.tpondemand.com/entity/7540
#### Linked PRs in other repos
n/a
#### Manual testing instructions  
New DM has been tested and approved.
New Room has been redesigned *and* rerouted. The behaviour has not been touched so that should not be any different. You should be on the lookout for anything that breaks the layout (especially in the text inputs), as well as anything that links to the "new-channel" view. (I counted 3: "Rooms +" button in main chat list, "Create a Room instead" in the new DM view, and "Create a Room +" in the bonus storage unlocks view).
#### Additional notes
UserPicker component, already an absolute mess, is now even more of a rat's nest! @seavan and I have a plan to refactor it.

----
### Reviewer

* Localisation.
* Architectural decisions.
* Performance.
* Comments.
* Big feature tech design documentation.
* Readability.
* DRY.
* Can we extract something to SDK?
* Are there any bicycles invented that should be replaced with existing SDK
* Are there unit and e2e tests added?

----
### Repository owner
